docs/03-declarative-layout.md

```diff
@@ -29,7 +28,116 @@ configurable rules.
 
 @DOC_layout_WIDGET_LIST@
 
-## Placing widgets
+### The different type of widget boxes (Wibox)
+
+Awesome API the word "wibox" to describe an area of the screen filled with
+widgets. There is many subvariants of wiboxes with specialized role such as
+widget bars or tooltips. All variants mostly share the same characteristics,
+but add some extra features to make those specialized widget boxes easier to
+work with.
+
+@DOC_awful_popup_wiboxtypes_EXAMPLE@
+
+The normal `wibox` is the base class for each of these type. It is extremely
+flexible and allows to place just about anything on the screen. However it
+requires a lot of repetitive boiler plate code to use directly. For example,
+the user need to compute the optimal size by hand or use `awful.placement`.
+
+The `awful.wibar` specialization allows to attach a `wibox` to a screen edge
+and prevent clients from using this area when tiled.
+
+The `awful.popup` allows to easily place widgets on the screen. It automatically
+resize itself to fit the optimal widget size. It also have helper properties
+and method to make it easy to place it on the screen. It both absolute
+positioning, relative positioning and manual positioning.
+
+The `awful.tooltip` is a very simple `wibox` that allows to display text next
+to an object such as the mouse.
+
+Finally, the `awful.titlebar`, while not technically a real `wibox`, act
+exactly the same way and allows to attach widgets on each side of the clients.
+
+## The different syntaxes to initiate widgets
+
+Awesome provides 2 totally different API access style to manage widgets. Both
+suit different use case and are execute exactly the same code under the hood.
+
+The **imperative** widget initialization is similar to QtWidgets, GTK and Win32.
+You create the object, the set the property and add the widget as a child to
+another already declared widget. It is quite simple to use but very verbose
+and full of boilerplate code. The imperative API also offers property both with
+accessors or directly. It is useful when creating highly dynamic layouts where
+widgets are added and removed over the course of their lifecycle.
+
+The **declarative** syntax resembles HTML style code
+written in JSON or YAML. The widgets instances are created automatically and
```

Comment:
HTML written in JSON or YAML?!?!?!?!?!?!?!?!?!?

Reply:
Okay:

The **declarative** syntax resembles HTML-style code in that an explicit nesting is present and is used to express "boxes containing boxes" (the box model). However, instead of requiring (heavy-weight) XML, a lighter JSON-like syntax is used.